db/src/main/kotlin/edu/artic/db/AppDataManager.kt

```diff
@@ -133,9 +139,25 @@ class AppDataManager @Inject constructor(
                             val result = progressDataState.result as ArticResult<*>
                             if (result.data.isNotEmpty() && result.data[0] is ArticExhibition) {
                                 appDatabase.exhibitionDao.clear()
-                                @Suppress("UNCHECKED_CAST")
-                                appDatabase.exhibitionDao.updateExhibitions((result as ArticResult<ArticExhibition>).data)
-
+                                /**
+                                 * Update the sort order of the exhibitions according to the ArticExhibitionCMS
+                                 */
+                                appDatabase.exhibitionCMSDao
+                                        .getAllCMSExhibitions()
+                                        .subscribe {
+                                            @Suppress("UNCHECKED_CAST")
+                                            val list = (result as ArticResult<ArticExhibition>).data
+                                            val mapExhibitionByID = HashMap<String, ArticExhibition>()
+
+                                            list.forEach {
+                                                mapExhibitionByID[it.id.toString()] = it
+                                            }
+
+                                            it.forEach {
+                                                mapExhibitionByID[it.id]?.order = it.sort
```

Comment:
no nested `it`. confusing here. use direct name on the outer object.

Reply:
Refactored.